### DISCUSSION TITLE
Is it possible to create just some markdown pages for welcome and tutorials?

### DISCUSSION THREAD

**Discussion by aglasencnik** [jsakamoto/BlazingStory#30] [Q&A]
Is it possible to create just some markdown pages for welcome and tutorials?
The reason is, so I could make a tutorial page on how to use my components library... and also so it wouldn't create a Docs page on it's own for that page...

**Comment by jsakamoto** [comment]
Hi @aglasencnik,
Yeah, it would be great to have such markdown pages instead of the default "Docs" page. Unfortunately, the "Blazing Story" doesn't have that ability now. I'd like to implement that ability someday.

**Reply by aglasencnik** [reply]
Hi @jsakamoto, I have some additional questions/problems....

- The Discord link doesn't seem to be working
- I tried making a BlazingStory storybook yesterday, and included quite some components, but haven't used any yet... And basically because of the amount of components (the library I am wrapping is quite big) it becomes really slow, and if I use debugging to try to debug a component, the browser crashes completely if I have a breakpoint anywhere (I am not even in the page which executes to the breakpoint) also I am on a very powerful laptop so I don't think it is a power issue
- I also tried publishing to github pages and it threw a million errors and each story got an "unhandled exception". When I looked it for some reason haven't uploaded any files which use _content?? Do I have to do anything special in that case?

- I got quite a few ideas what wold be very useful in this library, could I also become a contributor and make a pull request if I implemented anything? Some ideas:
- Custom themes
- Custom ordering of categories and stories (I would need this really bad)
- Markdown pages (and maybe actually use markdown???)

**Reply by aglasencnik** [reply]
Also how could I debug the component library?? I have breakpoints and tun in debug mode, but they are never caught.....

**Reply by jsakamoto** [reply]
@aglasencnik 
> Also how could I debug the component library?? I have breakpoints and tun in debug mode, but they are never caught.....

Could you tell me what type the story app is? Blazor Server or WebAssembly?

**Reply by jsakamoto** [reply]
@aglasencnik 
> I also tried publishing to github pages and it threw a million errors and each story got an "unhandled exception". When I looked it for some reason haven't uploaded any files which use _content?? Do I have to do anything special in that case?

Ah, your story app is a Blazor WebAssembly app, right? So, the most reasonable way to resolve this issue is to provide me with all of its source code and project.

**Reply by aglasencnik** [reply]
WebAssembly, so I could publish to github pages.... Would it be better if I used server? Or if I just use both, and make a shared library for the pages??? And test on the server one and publish the wasm one?

**Reply by aglasencnik** [reply]
Why would you need the source code? Also that isn't the biggest issue now... the issue now is how to debug the application....
I would send you the source code but the issue is that I took down the repository....

**Reply by aglasencnik** [reply]
Also what about the contributing?? Could I also make a contribution?

**Reply by jsakamoto** [reply]
> Why would you need the source code?

Because I have been really tired to reproduce such a problem on my side. In many case, the obviously problem on your side is not obvious on my side. I have ran into many cases that I couldn’t reproduce the problem people reported me except they provide a source code that can reproduce the problem.

**Reply by jsakamoto** [reply]
> Could I also make a contribution?

Basically yes, but please remember that I am considering making a big breaking change on the story syntax.

**Reply by aglasencnik** [reply]
Ok, I will create a new project and try to make a small minimal version, to try if it works.... and I'll come back to you

**Reply by aglasencnik** [reply]
@jsakamoto  Here: [https://github.com/aglasencnik/BlazorCarbonChartsPublishTest](https://github.com/aglasencnik/BlazorCarbonChartsPublishTest) is how I would Ideally like to have it working.... Currently it only should work in the Simple area chart story.... But it already doesn't work like supposed to, because the stories aren't shown.....
Could you please help me get it up and running?

**Reply by aglasencnik** [reply]
Hi, so I'm sorry but I have forgot a couple of things in the source code (I just made something fast as a demo....) basically I want to achieve having all the stories in the shared library, or at least have debugging capabilities in wasm project (then shared library for stories isn't necessery...)...
I fixed it now, and just put the stories in the wasm project for now also.... The component shows, but debugging doesn't work in https console mode....
How to reproduce: Just put a debugging point in the Base component I have in the main library BlazorCarbonCharts -> Components -> BaseCarbonChartComponent and the point won't be ever hit (at least on my machine)
Also after some time on one of the stories I get this error:
```
crit: Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Timeout
      Error: Timeout
          at waitFor (https://localhost:7281/_content/BlazingStory/Internals/Components/Preview/PreviewFrame.razor.js?v=1.0.0-preview.35:13:19)
          at async waitForIFrameReady (https://localhost:7281/_content/BlazingStory/Internals/Components/Preview/PreviewFrame.razor.js?v=1.0.0-preview.35:19:12)
          at async Module.navigatePreviewFrameTo (https://localhost:7281/_content/BlazingStory/Internals/Components/Preview/PreviewFrame.razor.js?v=1.0.0-preview.35:35:48)
Microsoft.JSInterop.JSException: Timeout
Error: Timeout
    at waitFor (https://localhost:7281/_content/BlazingStory/Internals/Components/Preview/PreviewFrame.razor.js?v=1.0.0-preview.35:13:19)
    at async waitForIFrameReady (https://localhost:7281/_content/BlazingStory/Internals/Components/Preview/PreviewFrame.razor.js?v=1.0.0-preview.35:19:12)
    at async Module.navigatePreviewFrameTo (https://localhost:7281/_content/BlazingStory/Internals/Components/Preview/PreviewFrame.razor.js?v=1.0.0-preview.35:35:48)
   at Microsoft.JSInterop.JSRuntime.<InvokeAsync>d__16`1[[Microsoft.JSInterop.Infrastructure.IJSVoidResult, Microsoft.JSInterop, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]].MoveNext()
   at Microsoft.JSInterop.JSObjectReferenceExtensions.InvokeVoidAsync(IJSObjectReference jsObjectReference, String identifier, Object[] args)
   at BlazingStory.Internals.Services.JSModule.InvokeVoidAsync(String identifier, Object[] args)
   at BlazingStory.Internals.Components.Preview.PreviewFrame.UpdatePreviewFrameUrlAsync()
   at BlazingStory.Internals.Components.Preview.PreviewFrame.OnParametersSetAsync()
   at Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle, ComponentState owningComponentState)
```

**Reply by aglasencnik** [reply]
Update: I just moved all the components the server app and the debugging works!!! But still I don't know how to move them to a different shared library to use both....

**Reply by aglasencnik** [reply]
Hi @jsakamoto so I found out I needed to add another assembly to detect the shared library... 
Also I got the github pages working, but I had to change the base path of both iframe and index with the sed command in actions (you can look in the repo I sent)...
But I still occasionally get the error which I pasted above, and chrome caching absolutely hates it haha... I have to go on different browser, then it works without a problem or in the incognito mode.

**Comment by jsakamoto** [comment]
> Also I got the github pages working, but I had to change the base path of both iframe and index with the sed command in actions (you can look in the repo I sent)...

@aglasencnik Please see the https://github.com/jsakamoto/BlazingStory/discussions/33

**Comment by jsakamoto** [comment]
> Is it possible to create just some markdown pages for welcome and tutorials?

Yeah, it would be great to have such markdown pages instead of the default "Docs" page. Unfortunately, the "Blazing Story" doesn't have that ability now. I'd like to implement that ability someday.

**Comment by jsakamoto** [comment]
>  I found out I needed to add another assembly to detect the shared library...

@aglasencnik I logged a task to improve the README into the issues: https://github.com/jsakamoto/BlazingStory/issues/34

**Comment by jsakamoto** [comment]
> at least have debugging capabilities in wasm project (then shared library for stories isn't necessery...)...

@aglasencnik Please see https://github.com/jsakamoto/BlazingStory/discussions/35

**Comment by lgk-bsw** [comment]
I created myself a "workaround". I added a story for a Blazor component I named "**Welcome.razor**" with this content:

```razor
<!-- This is necessary, so links won't open inside the iframe:  -->
<base target="_top">

<div class="container">
    <md-block>
        # Introduction
        Blue Blazor is an adaptation of [Blue React](https://bruegmann.github.io/blue-react) for Blazor.

        ## Stylesheet
        You can use the stylesheet of Blue React by adding the following line to the head of your HTML:

        ```html
        &lt;link rel="stylesheet" href="https://unpkg.com/blue-react@latest/dist/style.min.css" /&gt;
        ```

        ## References
        This site is made with [BlazingStory](https://github.com/jsakamoto/BlazingStory).
    </md-block>
</div>

<script type="module">
    import { MarkdownBlock, URLs } from "https://md-block.verou.me/md-block.js"
    URLs.Prism = "https://unpkg.com/prismjs@1.29.0/prism.js"
    URLs.PrismCSS = "https://unpkg.com/prismjs@1.29.0/themes/prism.min.css"
</script>
```

This uses Lea Verou's Web Component [md-block](https://md-block.verou.me/) for Markdown support. 

**Welcome.stories.razor** looks like this:
```razor
@attribute [Stories("Area/Welcome")]

<Stories TComponent="Welcome">

    <Story Name="Introduction">

        <Arguments>
        </Arguments>

        <Template>
            <Welcome @attributes="context.Args" />
        </Template>

    </Story>

</Stories>
```

I put it in the section "Area", so it will appear on top of "Components" in the navigation bar.

**Reply by aglasencnik** [reply]
Hi, this is great! Thank you very very very much. Maybe I could add it as a type of story?? So instead of story it is a documentation page in and then make a pull request to the repository, so it would get supported officially?? Maybe instead of filename.stories.razor it could be filename.docs.razor ??? 
Would you find that as useful?

**Reply by lgk-bsw** [reply]
I'm glad you find my solution useful. Official support for Markdown would be awesome. 
I'm not sure if `*.docs.razor` would be the best format for naming them. Is that the way to do it in Storybook?

**Comment by AlexHedley** [comment]
The [Fluent UI Blazor](https://github.com/microsoft/fluentui-blazor/) project has method of using displaying [markdown](https://daringfireball.net/projects/markdown/syntax, it has a dependency on [markdig](https://github.com/xoofx/markdig)

I wrote up an [article](https://alexhedley.com/blog/posts/blazor-fluentui-markdownsection) on how to add the Component to your own app(s).

Would this be something you'd consider?

**Comment by Snellingen** [comment]
I have created a PR that would support this with custom pages: 
https://github.com/jsakamoto/BlazingStory/pull/67

Here you could use md-block or other libraries for adding markdown as it will simple add a custom razor page to your storybook
